components/brave_sync/brave_sync_service_unittest.cc

```diff
@@ -246,6 +250,38 @@ class BraveSyncServiceTest : public testing::Test {
   base::ScopedTempDir temp_dir_;
 };
 
+class BraveSyncServiceTestDelayedLoadModel : public BraveSyncServiceTest {
+ public:
+  void PrepareModelFactory() override {
+    BookmarkModelFactory::GetInstance()->SetTestingFactory(
+        profile(),
+        base::BindRepeating(&BraveSyncServiceTestDelayedLoadModel::BuildModel));
+  }
+  void ModelDoneLoading() {
+    model()->DoneLoading(std::move(bookmark_load_details_));
+  }
+ private:
+  static std::unique_ptr<KeyedService> BuildModel(content::BrowserContext* context) {
```

Comment:
why is it a static function here? it access `bookmark_load_details_` and it is even in `private` section

Reply:
My, bad. I was sure `KeyedServiceFactory::SetTestingFactory` takes plain function as a parameter, but it takes `base::RepeatingCallback`. Fixed that, thanks for pointing.